demo-simple/src/main/java/com/novoda/downloadmanager/demo/DatabaseClonerFactory.java

```diff
@@ -0,0 +1,42 @@
+package com.novoda.downloadmanager.demo;
+
+import android.content.Context;
+import android.content.res.AssetManager;
+import android.os.Handler;
+import android.os.Looper;
+
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
+
+final class DatabaseClonerFactory {
+
+    private DatabaseClonerFactory() {
+        // Uses static factory method.
+    }
+
+    static VersionOneDatabaseCloner databaseCloner(Context context, final VersionOneDatabaseCloner.CloneCallback cloneCallback) {
+        String originalDatabaseLocation = context.getFilesDir().getParentFile().getAbsolutePath() + "/databases/downloads.db";
+        Executor executor = Executors.newSingleThreadExecutor();
+        AssetManager assets = context.getAssets();
+        final Handler mainThreadHandler = new Handler(Looper.getMainLooper());
+
+        VersionOneDatabaseCloner.CloneCallback mainThreadReportingCloneCallback = new VersionOneDatabaseCloner.CloneCallback() {
```

Comment:
Decided to hide this callback on the `main thread` here to prevent unnecessary noise in the `VersionOneDatabaseCloner` class. Also reduces the amount of code in the `activity`.